src/Microsoft.AspNet.Mvc.Core/ResponseCacheAttribute.cs

```diff
@@ -0,0 +1,105 @@
+    {
+        // A nullable-int cannot be used as an Attribute parameter.
+        // Hence this nullable-int is present to back the Duration property.
+        private int? _duration;
+        
+        /// <summary>
+        /// Gets or sets the duration in seconds for which the response is cached.
+        /// </summary>
+        public int Duration
+        { 
+            get
+            {
+                if (_duration == null)
+                {
+                    return 0;
+                }
+
+                return (int)_duration;
+            }
+            set
+            {
+                _duration = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the location where the data from a particular URL must be cached.
+        /// </summary>
+        public ResponseCacheLocation Location { get; set; }
+
+        /// <summary>
+        /// Gets or sets the value which determines whether the data should be stored or not.
+        /// </summary>
+        public bool NoStore { get; set; }
+
+        /// <summary>
+        /// Gets or sets the value for the Vary response header.
+        /// </summary>
+        public string VaryByHeader { get; set; }
+
+        // <inheritdoc />
+        public override void OnActionExecuting([NotNull] ActionExecutingContext context)
+        {
+            var headers = context.HttpContext.Response.Headers;
+
+            if (!string.IsNullOrEmpty(VaryByHeader))
+            {
+                headers.Set("Vary", VaryByHeader);
+            }
+
+            if (NoStore)
+            {
+                headers.Set("Cache-control", "no-store");
+                // Cache-control: no-store, no-cache is valid.
+                if (Location == ResponseCacheLocation.None)
+                {
+                    headers.Append("Cache-control", "no-cache");
+                    headers.Set("Pragma", "no-cache");
+                }
+            }
+            else
+            {
+                string cacheControlValue = null;
+                switch (Location)
+                {
+                    case ResponseCacheLocation.Any:
+                        cacheControlValue = "public";
+                        break;
+                    case ResponseCacheLocation.Client:
+                        cacheControlValue = "private";
+                        break;
+                    case ResponseCacheLocation.None:
+                        cacheControlValue = "no-cache";
+                        headers.Set("Pragma", "no-cache");
+                        break;
+                }
+
+                if (_duration != null)
+                {
+                    cacheControlValue = string.Format(
+                        CultureInfo.InvariantCulture,
+                        "{0}{1}max-age={2}",
+                        cacheControlValue,
+                        cacheControlValue != null? "," : null,
+                        Duration.ToString());
```

Comment:
either specify `Duration.ToString(CultureInfo.InvariantCulture)` or just take a away the `.ToString()` and string.format will do it for you.